redfish/provider/resource_redfish_virtual_media.go

```diff
@@ -1,26 +1,456 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/stmcginnis/gofish"
+	"github.com/stmcginnis/gofish/redfish"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource = &virtualMediaResource{}
+)
+
+// NewVirtualMediaResource is a helper function to simplify the provider implementation.
+func NewVirtualMediaResource() resource.Resource {
+	return &virtualMediaResource{}
+}
+
+// virtualMediaResource is the resource implementation.
+type virtualMediaResource struct {
+	p *redfishProvider
+}
+
+const (
+	serviceError = "service error"
+	mountError   = "Couldn't mount Virtual Media"
+)
+
+// Configure implements resource.ResourceWithConfigure
+func (r *virtualMediaResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	r.p = req.ProviderData.(*redfishProvider)
+}
+
+// Metadata returns the resource type name.
+func (*virtualMediaResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "virtual_media"
+}
+
+// VirtualMediaSchema defines the schema for the resource.
+func VirtualMediaSchema() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"redfish_server": schema.SingleNestedAttribute{
+			MarkdownDescription: "Redfish Server",
+			Description:         "Redfish Server",
+			Required:            true,
+			Attributes:          RedfishServerSchema(),
+		},
+		"id": schema.StringAttribute{
+			MarkdownDescription: "ID of the virtual media resource",
+			Description:         "ID of the virtual media resource",
+			Computed:            true,
+		},
+		"image": schema.StringAttribute{
+			Required:            true,
+			Description:         "The URI of the remote media to attach to the virtual media",
+			MarkdownDescription: "The URI of the remote media to attach to the virtual media",
+		},
+		"inserted": schema.BoolAttribute{
+			Computed:            true,
+			Description:         "Describes whether virtual media is attached or detached",
+			MarkdownDescription: "Describes whether virtual media is attached or detached",
+		},
+		"transfer_method": schema.StringAttribute{
+			Optional:            true,
+			Computed:            true,
+			Description:         "Indicates how the data is transferred",
+			MarkdownDescription: "Indicates how the data is transferred",
+			Validators: []validator.String{
+				stringvalidator.OneOf([]string{
+					"Stream",
+					"Upload",
+				}...),
+			},
+		},
+		"transfer_protocol_type": schema.StringAttribute{
```

Comment:
`transfer_protocol_type` and `transfer_method` are Optional and Computed. Are there default values?
I see that we are using code like 
```
virtualMediaConfig := redfish.VirtualMediaConfig{
		Image:                image,
		Inserted:             plan.Inserted.ValueBool(),
		TransferMethod:       plan.TransferMethod.ValueString(),
		TransferProtocolType: plan.TransferProtocolType.ValueString(),
		WriteProtected:       plan.WriteProtected.ValueBool(),
	}
```
Please check if these will work correctly without default values.

Reply:
tested, as this are optional plus computed it shows them as "known after apply" and creates the resource successfully.

  ```
+ resource "redfish_virtual_media" "vm" {
      + id                     = (known after apply)
      + image                  = "http://linuxlib.us.dell.com/pub/redhat/RHEL8/8.8/BaseOS/x86_64/os/images/efiboot.img"
      + inserted               = (known after apply)
      + redfish_server         = {
          + endpoint = "https://10.226.197.128"
          + password = (sensitive value)
          + user     = "root"
        }
      + transfer_method        = (known after apply)
      + transfer_protocol_type = (known after apply)
      + write_protected        = true
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

redfish_virtual_media.vm["my-server-1"]: Creating...
redfish_virtual_media.vm["my-server-1"]: Creation complete after 6s [id=/redfish/v1/Managers/iDRAC.Embedded.1/VirtualMedia/RemovableDisk]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

```

Reply:
then fine